nats-concepts/jetstream/headers.md

```diff
@@ -0,0 +1,55 @@
+| `Nats-Expected-Last-Sequence`         | Used to apply optimistic concurrency control at the stream-level. The value is the last expected sequence and the server will reject a publish if the current sequence does not match.                            | `328`                                  | 2.2.0   |
+| `Nats-Expected-Last-Subject-Sequence` | Used to apply optimistic concurrency control at the subject-level. The value is the last expected sequence and the server will reject a publish if the current sequence does not match for the message's subject. | `38`                                   | 2.3.1   |
+| `Nats-Rollup`                         | Used to apply a purge of all prior messages in the stream or at the subject-level.                                                                                                                                | `all` for stream, `sub` for subject    | 2.6.2   |
+
+## RePublish
+
+Headers set messages that are republished.
+
+| Name                 | Description                                                                                                            | Example                       | Version |
+| :------------------- | :--------------------------------------------------------------------------------------------------------------------- | :---------------------------- | :------ |
+| `Nats-Stream`        | Name of the stream the message was republished from.                                                                   | `Nats-Stream: my-stream`      | 2.8.3   |
+| `Nats-Subject`       | The original subject of the message.                                                                                   | `events.mouse_clicked`        | 2.8.3   |
+| `Nats-Time-Stamp`    | The original timestamp of the message.                                                                                 | `2023-08-23T19:53:05.762416Z` | 2.8.3   |
+| `Nats-Sequence`      | The original sequence of the message.                                                                                  | `193`                         | 2.8.3   |
+| `Nats-Last-Sequence` | The last sequence of the message having the same subject, otherwise zero if this is the first message for the subject. | `190`                         | 2.8.3   |
+
+## Sources
+
+Headers that are implicitly added to messages sourced from other streams.
+
+| Name                 | Description                                                 | Example     | Version |
+| :------------------- | :---------------------------------------------------------- | :---------- | :------ |
+| `Nats-Stream-Source` | Specifies the stream a message sourced from, if applicable. | `my-stream` | 2.2.0   |
```

Comment:
Specifies the original stream name, the subject and the sequence number plus the subject filter and destination transform of the message being sourced.

Reply:
Updated to use this definition.